src/test/scala/mesosphere/marathon/core/task/tracker/impl/steps/UpdateTaskTrackerStepImplTest.scala

```diff
@@ -0,0 +1,201 @@
+package mesosphere.marathon.core.task.tracker.impl.steps
+
+import mesosphere.marathon.MarathonSchedulerDriverHolder
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.state.{ PathId, Timestamp }
+import mesosphere.marathon.tasks.{ TaskIdUtil, TaskTracker }
+import mesosphere.marathon.test.Mockito
+import org.apache.mesos.Protos.{ SlaveID, TaskState, TaskStatus }
+import org.apache.mesos.SchedulerDriver
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ FunSuite, GivenWhenThen, Matchers }
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+class UpdateTaskTrackerStepImplTest extends FunSuite with Matchers with ScalaFutures with Mockito with GivenWhenThen {
```

Comment:
I thought the original task was to write tests for the new logic. I see changed logic, removed tests and the only added test is not for review...
Do you want me to merge or is this just WIP?